src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_archimonde.cpp

```diff
@@ -143,35 +143,36 @@ struct npc_doomfire_spirit : public ScriptedAI
 {
     npc_doomfire_spirit(Creature* creature) : ScriptedAI(creature){ }
 
+    float m_fAngle = urand(0, M_PI * 2);
+
     void Reset() override
     {
         scheduler.CancelAll();
         ScheduleTimedEvent(0s, [&] {
-            me->GetMotionMaster()->MovePoint(NEAR_POINT, DoomfireMovement(me->GetPosition()));
-            }, 1500ms);
+            float nextOrientation = Position::NormalizeOrientation(me->GetOrientation() + irand(-1, 1) * 0.785402f);
+            Position pos = GetFirstRandomAngleCollisionPosition(8.f, nextOrientation); // both orientation and distance verified with sniffs
+            me->NearTeleportTo(pos.GetPositionX(), pos.GetPositionY(), pos.GetPositionZ(), nextOrientation);
+            }, 1600ms);
+
+        m_fAngle = urand(0, M_PI * 2);
     }
 
-    Position DoomfireMovement(Position mePos)
+    Position GetFirstRandomAngleCollisionPosition(float dist, float angle)
     {
-        float angle = mePos.GetOrientation();
-        float distance = 100.0f;
-        float newAngle = angle + ((rand() % 181) - 90) * M_PI / 180;
-        float x = mePos.GetPositionX() + distance * cos(newAngle);
-        float y = mePos.GetPositionY() + distance * sin(newAngle);
-
-        Position targetPos = Position(x, y, me->GetPositionZ());
-        return targetPos;
+        Position pos;
+        for (uint32 i = 0; i < 10; ++i)
```

Comment:
amount of attempts is just arbitrary, if I understand correctly? fine by me, but would rather not see magic numbers so best to use some constant instead of 10

Reply:
If you're dead set on it, I can swap it for a constant, but given that its usage is (in my opinion at least) pretty darn hard to misinterpret, I wouldn't really call it a magic number per se.

Reply:
What is it based on then? The number does not really explain itself. Hence why I called it a magic number

Reply:
I dont mind this. It just says we want to run the for loop as long as i < 10